sqlglot/dialects/tsql.py

```diff
@@ -16,6 +16,8 @@
 from sqlglot.time import format_time
 from sqlglot.tokens import TokenType
 
+E = t.TypeVar("E", bound="exp.Expression")
```

Comment:
we have this in a couple places, can we move this to like typing or something

Reply:
Yeah I can do that

Reply:
Done